sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -114,7 +114,7 @@ private[sql] object DataFrame {
 @Experimental
 class DataFrame private[sql](
     @transient val sqlContext: SQLContext,
-    @DeveloperApi @transient val queryExecution: SQLContext#QueryExecution) extends Serializable {
+    @DeveloperApi @transient val queryExecution: QueryExecution) extends Serializable {
```

Comment:
The signature of the function to retrieve it has changed though.  I think that we should rename this parameter and add a method that constructs a new instance of the inner subclass to return with the name `queryExecution`.  I think its worth rerunning the optimizer to not have to recompile all our perf tools.